### ISSUE TITLE
AWS - NodeJS v8.x Deprecation

### ISSUE THREAD

**Issue by krampus-nuggets** [pimp-my-book/serverless-graphql-nodejs-starter#10]
> What’s happening?

The Node community has decided to end support for Node.js 8.x on December 31, 2019 [1]. From this date forward, Node.js 8.x will stop receiving bug fixes, security updates, and/or performance improvements. To ensure that your new and existing functions run on a supported and secure runtime, language runtimes that have reached their EOL are deprecated in AWS [2].

For Node.js 8.x, there will be 2 stages to the runtime deprecation process:

1. Disable Function Create – Beginning January 6, 2020, customers will no longer be able to create functions using Node.js 8.10

2. Disable Function Update – Beginning February 3, 2020, customers will no longer be able to update functions using Node.js 8.10

After this period, both function creation and updates will be disabled permanently. However, existing Node 8.x functions will still be available to process invocation events.

> What do I need to do?

We encourage you to update all of your Node.js 8.10 functions to the newer available runtime version, Node.js 10.x[3]. You should test your functions for compatibility with the Node.js 10.x language version before applying changes to your production functions.

**Comment by AmoDinho** [comment]
This is massive bruv.

I will need to figure out how to migrate this  existing repo to Node 10.x and all our other production functions.

**Comment by krampus-nuggets** [comment]
Spit balling here:

Couldn't you explicitly define the currently/already created Cognito, S3, etc. resources in their serverless.yml & Other YAML config files and just update the runtime to nodejs v10.x. That way you'd update the runtime and connect the needed resources without Serverless redeploying with newly created resources?

TL;DR

1. Update runtime to v10.x in serverless.yml
2. Explicitly define other needed resources in their own config files such as **cognito-user-pool.yml**

**Comment by AmoDinho** [comment]
Yeah I agree with ya, the issue I foresee is that some of the dependencies might not have been upgraded to support Node 10.x however this is just a hunch. 

But atleast we only have two projects with 8.x runtime lambdas so it should not be too bad.
